File: app/src/main/java/com/mgaetan89/showsrage/presenter/LogPresenter.java
package com.mgaetan89.showsrage.presenter;

import android.support.annotation.ColorRes;
import android.support.annotation.Nullable;

import com.mgaetan89.showsrage.helper.DateTimeHelper;
import com.mgaetan89.showsrage.model.LogEntry;

public class LogPresenter {
	@Nullable
	private LogEntry logEntry;

	public LogPresenter(@Nullable LogEntry logEntry) {
		this.logEntry = logEntry;
	}

	public CharSequence getDateTime() {
		if (this.logEntry == null) {
			return "";
		}

		return DateTimeHelper.INSTANCE.getRelativeDate(this.logEntry.getDateTime(), "yyyy-MM-dd hh:mm:ss", 0);
	}

	@ColorRes
	public int getErrorColor() {
		if (this.logEntry == null) {
			return android.R.color.white;
		}

		return this.logEntry.getErrorColor();
	}

	public String getErrorType() {
		if (this.logEntry == null) {
			return "";
		}

		return this.logEntry.getErrorType();
	}

	public String getMessage() {
		if (this.logEntry == null) {
			return "";
		}

		return this.logEntry.getMessage().trim();
	}
}

File: app/src/main/kotlin/com/mgaetan89/showsrage/presenter/LogPresenter.kt
package com.mgaetan89.showsrage.presenter

import android.support.annotation.ColorRes
import com.mgaetan89.showsrage.extension.humanize
import com.mgaetan89.showsrage.extension.toRelativeDate
import com.mgaetan89.showsrage.model.LogEntry

class LogPresenter(private val logEntry: LogEntry?) {
	fun getDateTime(): CharSequence? {
		if (!this.isLogEntryValid()) {
			return ""
		}

		return this.logEntry?.dateTime.toRelativeDate("yyyy-MM-dd hh:mm:ss", 0)
	}

	@ColorRes
	fun getErrorColor() = if (this.isLogEntryValid()) this.logEntry!!.getErrorColor() else android.R.color.white

	fun getErrorType() = if (this.isLogEntryValid()) this.logEntry!!.errorType ?: "" else ""

	fun getGroup() = if (this.isLogEntryValid()) this.logEntry!!.group?.humanize() else null

	fun getMessage() = if (this.isLogEntryValid()) this.logEntry!!.message.trim() else ""

	internal fun isLogEntryValid() = this.logEntry != null && this.logEntry.isValid
}
